### PR TITLE
Javalab Neighborhood: Support painter initialize and move

### PR DESCRIPTION
**Pull request by molly-moen**
This PR adds support for initializing and moving multiple painters within the neighborhood. The level can optionally define a start painter location, and this PR adds an additional allthethings level which does not include a start location. This PR also adds the constants for the signal types for other painter actions (such as paint, remove paint) to be implemented as a follow-up.

Samples of neighborhood in action

Level that defines a start painter (note painter disappears on compile, this is a curriculum requirement):

https://user-images.githubusercontent.com/33666587/119056415-84818d00-b97f-11eb-93ad-a2932f4a3148.mp4

Level that does not define a start painter:

https://user-images.githubusercontent.com/33666587/119056446-9105e580-b97f-11eb-9d2b-141e23122a68.mp4

Note: This PR depends on a new version of Javabuilder that has not yet been deployed.

## Links

- spec: [Neighborhood Engineering Plan](https://docs.google.com/document/d/1SqSLogXsOKJD8QexbZAbxCpUGRdNP9Yj7zW8tN6oADI/edit)
- jira ticket: [CSA 293](https://codedotorg.atlassian.net/browse/CSA-293)

## Testing story
Tested locally. Most changes here are specific to neighborhood, which is not in use yet. Maze upgrade involved very few possibilities of impacting other maze levels, as only changes were additional functions or if statements to ensure neighborhood didn't hit null errors. I did some spot testing on other various Maze levels and found no change in behavior.

## Deployment strategy
We will want to deploy the latest Javabuilder changes before merging this PR.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/javalab/constants.js [hunk 1]
```diff
@@ -24,5 +24,22 @@ export const JavabuilderExceptionType = {
 };
 
 export const NeighborhoodSignalType = {
-  MOVE: 'MOVE'
+  MOVE: 'MOVE',
+  INITIALIZE_PAINTER: 'INITIALIZE_PAINTER',
```
Comment (jmkulwik) [inline apps/src/javalab/constants.js:30]: NIT: Add comments before the first two signals as well.

File: dashboard/config/scripts/levels/Allthethings Java Lab 3.level [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+  <config><![CDATA[{
+  "properties": {
+    "encrypted": "false",
+    "instructions_important": "false",
+    "teacher_markdown": "Teachers should see this extra info.",
+    "long_instructions": "This is a sample Java Lab Neighborhood level.",
+    "mini_rubric": "false",
+    "hide_share_and_remix": "false",
+    "csa_view_mode": "neighborhood",
+    "serialized_maze": [
```
Comment (jmkulwik) [inline dashboard/config/scripts/levels/Allthethings Java Lab 3.level:11]: Does this need to be stringified? i.e. the formatting is different for Allthethings Java Lab 1.level

File: apps/src/javalab/Neighborhood.js [hunk 3]
```diff
@@ -41,31 +38,74 @@ export default class Neighborhood {
...
+      timeout = this.lastSignalEndTime - Date.now();
+      if (timeout < 0) {
+        timeout = 0;
+      }
+      this.lastSignalEndTime = this.lastSignalEndTime + totalSignalTime;
+    } else {
+      this.lastSignalEndTime = Date.now() + totalSignalTime;
+    }
+    timeoutList.setTimeout(
+      () => this.getMazeCommand(signal, timeForSignal),
```
Comment (jmkulwik) [inline apps/src/javalab/Neighborhood.js:64]: Nice! This will make it straightforward to add more maze commands.

**Review by jmkulwik — APPROVED** [review]
Couple comments, but not blocking. LGTM!

File: dashboard/config/scripts/levels/Allthethings Java Lab 3.level (hunk 2 above)
Comment (molly-moen) [reply dashboard/config/scripts/levels/Allthethings Java Lab 3.level:11]: it actually supports both. This is what you end up with if you clone an existing level with a serialized maze, and the parsing on Javalab supports either a string or an array.

File: apps/src/javalab/constants.js (hunk 1 above)
Comment (molly-moen) [reply apps/src/javalab/constants.js:30]: done!

**Pull request merged**